File: python-version/LC812/largest_triangle_area.py
from typing import List


class Solution:
    @staticmethod
    def largestTriangleArea(points: List[List[int]]) -> float:
        max_area, n = 0, len(points)
        for i in range(n):
            for j in range(i+1, n):
                for k in range(j+1, n):
                    area = 0.5 * abs(
                        points[i][0] * (points[j][1] - points[k][1]) +
                        points[j][0] * (points[k][1] - points[i][1]) +
                        points[k][0] * (points[i][1] - points[j][1])
                    )
                    max_area = max(area, max_area)
        return max_area

File: go-version/leetcode/LC812/largest_triangle_area.go
package LC812

import "math"

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/9/27 下午10:46
 * @FilePath: largest_triangle_area
 * @Description:
 */

func largestTriangleArea(points [][]int) float64 {
	var maxArea float64
	var n = len(points)
	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			for k := j + 1; k < n; k++ {
				area := 0.5 * math.Abs(
					float64(points[i][0]*(points[j][1]-points[k][1])+
						points[j][0]*(points[k][1]-points[i][1])+
						points[k][0]*(points[i][1]-points[j][1])),
				)
				maxArea = max(maxArea, area)
			}
		}
	}
	return maxArea
}
